storage/storage-interface/src/lib.rs

```diff
@@ -638,31 +641,89 @@ pub trait DbWriter: Send + Sync {
     }
 }
 
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum FastSyncStatus {
+    UNKNOWN,
+    STARTED,
+    FINISHED,
+}
+
 #[derive(Clone)]
 pub struct DbReaderWriter {
     pub reader: Arc<dyn DbReader>,
     pub writer: Arc<dyn DbWriter>,
+    // This is updated during StateSync bootstrapping.
+    pub fast_sync_status: Arc<RwLock<FastSyncStatus>>,
```

Comment:
Silly question: why can we not place this in the `FastSyncStorageWrapper` implementation itself? Putting it here feels off to me, i.e., we're leaking implementation details. Theoretically, `reader` and `writer` both take in interfaces, which should be `FastSyncStorageWrapper` underneath, right?

Reply:
Actually, I see that it's already in there. Why do we need it here? 🤔

Reply:
DbReaderWriter is the `interface` struct, which state-sync and other components use to access storage. State-sync is not supposed to directly access underlying FastSyncStorageWrapper's fields.

Reply:
Still not following -- will sync offline 😄